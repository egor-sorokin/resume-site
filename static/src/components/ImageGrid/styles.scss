@import '../../styles/base/_vars.scss';

.image-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;

  .item {
    min-width: 33%;
    min-height: 265px;
    height: 100%;

    padding: 20px 40px;

    background-color: $dune;
    box-shadow: -4px -4px 12px 1px black;
  }
}
