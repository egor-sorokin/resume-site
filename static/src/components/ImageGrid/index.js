import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const ImageGrid = ({gridData}) => {
  const component = gridData.map(item => (
    <li
      key={item.id}
      className="image-grid__item item text-c-mercury">
      <a
        href={item.url}
        target="_blank"
        className="item__image image"
        style={{ backgroundImage: `url(${item.image})` }}
      />
    </li>
  ));

  return (
    <ul className="image-grid">
      {component}
    </ul>
  );
};

ImageGrid.propTypes = {
  gridData: PropTypes.instanceOf(Array),
};

ImageGrid.defaultProps = {
  gridData: [],
};

export default ImageGrid;
