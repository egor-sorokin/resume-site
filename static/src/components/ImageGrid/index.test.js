// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import ImageGrid from './index';


describe('<ImageGrid />', () => {
  let component;
  const gridData = [
    {
      id: 1,
      label: 'title',
      url: '',
      image: 'assets/exp_name.jpg'
    },
    {
      id: 2,
      label: 'title',
      url: 'test.url',
      image: 'assets/exp_name.jpg'
    },
    {
      id: 3,
      label: 'title',
      url: 'test.url',
      image: 'assets/exp_name.jpg'
    },
    {
      id: 4,
      label: 'title',
      url: '',
      image: 'assets/exp_name.jpg'
    }
  ];


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<ImageGrid />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 4 components', () => {
      expect(component.children()).toHaveLength(0);
    });
  });


  describe('props tests', () => {
    beforeEach(() => {
      component = shallow(<ImageGrid gridData={gridData}/>);
    });


    it('renders 2 image with link', () => {
      expect(component.find('a')).toHaveLength(2);
    });


    it('renders 2 image without link', () => {
      expect(component.find('div')).toHaveLength(2);
    });
  });
});
