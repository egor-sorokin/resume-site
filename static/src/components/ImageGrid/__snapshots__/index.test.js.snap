// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageGrid /> base tests matches snapshot 1`] = `
<ul
  className="image-grid"
/>
`;
